apps/events.md

```diff
@@ -131,6 +135,106 @@ cart: {
     _links: null
 }
 ```
+
+## Product:click event
+
+To make use of this event, you need to add the following snippet to your code:
+
+```js
+if (window.eComEventTarget) {
+  window.eComEventTarget.addEventListener('product:click', function (event) {
+    console.log('product click', {
+      type: event.detail.type,
+      detail: event.detail.detail,
+      product: event.detail.product,
+      productIndex: event.detail.productIndex,
+    })
+  })
+}
+```
+
+You'll receive the following information:
+
+```
+detail: "Cherry Jam"
+product: {
+    availabilityText: "Available"
+    available: true
+    basePrice: {refQuantity: {…}, refPrice: {…}, formatted: "1 m³ = €0,12", quantity: {…}}
+    bulkPrices: null
+    conditionMicrodata: "NewCondition"
+    customAttributes: [{…}]
+    deliveryPeriod: "2-3"
+    deliveryPeriodUnit: "DAYS"
+    description: null
+    energyLabel: null
+    energyLabelSourceFile: null
+    gtin: 7501054530107
+    hasCrossSelling: false
+    hasStockLevel: true
+    hasVariations: false
+    href: "/p/homemade-cherry-jam"
+    image: null
+    isVariationMaster: false
+    isVariationProduct: false
+    isVisible: true
+    links: (5) [{…}, {…}, {…}, {…}, {…}]
+    listPrice: null
+    lowestPrice: null
+    mainCategoryId: "5954B711-E377-2A90-C400-D5809AB3F62B"
+    manufacturer: null
+    manufacturerPrice: null
+    metaDescription: ""
+    name: "Homemade Cherry Jam"
+    onStock: true
+    outOfStock: false
+    pickupPeriod: ""
+    price: {taxType: "NET", formatted: "€40,00", amount: 40, currency: "EUR"}
+    productDataSheet: null
+    productId: "5954B706-E701-F357-A52D-D5809AB3F606"
+    productVariationSelection: null
+    productVariationValues: ""
+    sku: "1007"
+    slideshow: [{…}, {…}, {…}, {…}]
+    slug: "homemade-cherry-jam"
+    stockLevelClass: "in"
+    stockLevelMicrodata: "InStock"
+    title: "Homemade Cherry Jam"
+    variationMaster: null
+    variations: null
+    vatNote: "components.productComponent.priceExclusiveVat"
+    warnStock: false
+}
+productIndex: 1
+type: "search"
+```
```

Comment:
I‘m not sure if it is good to have these kind of examples for each event, especially because the `console.log` example pattern is different or follows a different style throughout the examples.

The question that comes to my mind is: Is it important to list an example of the complete event details?

If it is enough to show _how to subscribe to these events_ and to tell the reader/app developer that they can _inspect/look at `event.detail`_ when they try out or develop their app, then only one example for all events could maybe also work.

Something like this (of course wording could be completely different):

```md
## Subscribing to an event

You can use the following code to subscribe to any event that is listed above.
Available information for each event can be found at the `event` property
"detail" in the event handler.

In this example, `event.detail` are output to the web browser‘s console.
Replace `EVENT_NAME` with the event you want to subscribe to, for example
"product".
```

```js
if (window.eComEventTarget) {
  window.eComEventTarget.addEventListener('EVENT_NAME', function (event) {
    // Do something with the event. All event information is in `event.detail`.
    console.log(event.detail)
  })
}
```

---

A problem is that for some events, the information is of type `Immutable.Map` that not every developer will know about, whereas for other events the information is in plain JavaScript object form. This mixture is not ideal in the first place (better would be if it is consistent). Issues I can think of:

- It _could_ lead to problems once real apps are using this storefront event API. Because if we change the types later on then it could be a breaking change for these apps. So a consistent format from the start would be best. Maybe we should make it consistent rather soon? So that we don‘t need to explain `Immutable.Map` or maintain it in the future.
- Now that there are plain JavaScript as well as `Immutable.Map` objects throughout the events, maybe we should document it on this page. [The "Type" information from the epages-ui document](https://github.com/ePages-de/epages-ui/blob/78c6cca10c5ae740f7719f942b0113ff556ace03/docs/javascript_events.md) that was now omitted on the table on this page for a good reason could give a hint. Still, that doesn‘t tell the developer what `Immutable.Map`s are or how to use them (but one could look it up). Also, for this, the examples for each type of event could help (which would speak against having only one example that documents how to subscribe to the events).

---

In any case, I think it would be good to at least make the examples consistent in their format (`console.log`). I could help with that. Maybe it would be easier to have a call about these topics if you agree with some of these points.